prov/shm/src/smr_progress.c

```diff
@@ -322,6 +323,13 @@ static int smr_progress_iov(struct smr_cmd *cmd, struct iovec *iov,
 		goto out;
 	}
 
+	xpmem = &smr_peer_data(ep->region)[cmd->msg.hdr.id].xpmem;
+	if (ep->region->xpmem_cap_self == SMR_VMA_CAP_ON &&
+	    xpmem->cap == SMR_VMA_CAP_ON) {
+		p2p_type = FI_SHM_P2P_XPMEM;
+		user_data = xpmem;
```

Comment:
probably can just pass in xpmem directly to the p2p_copy function to eliminate this local variable and the if/else

Reply:
I implemented this by adding a variable in the ep, which defaults to CMA and gets set to XPMEM if it's enabled. use_xpmem should remain a bool. It's more consistent with the rest of the other env vars: use_dsa_sar and disable cma.